### ISSUE TITLE
Cannot like a blog post with mastodon account

### ISSUE THREAD

**Issue by NilsRenaud** [Plume-org/Plume#581]
I tried to like a blog post on fediverse.blog with my mastodon account on m.g3l.org : NilsRenaud@m.g3l.org.
And I got an internal error.
I guess it should work since mastodon & plume use ActivityPub right ?

- **Plume version:** 0.3.0

**Comment by BaptisteGelez** [comment]
Yes, it should work.

**Comment by marek-lach** [comment]
Make sure that you do not have javascript disabled in your browser.

Otherwise, can you provide some screenshots as to what exactly happens and when?

I wanted to like a test post from fediverse.blog via the m.g3l.org instance, and had no problem at all.

<img width="1280" alt="Screenshot 2019-05-15 at 19 42 51" src="https://user-images.githubusercontent.com/45913977/57797313-6f4b4580-774a-11e9-8bb5-d26187aeff69.png">

If on fediverse.blog you click the heart icon, you'll get redirected and need to log in with your Mastodon instance:

<img width="1227" alt="Screenshot 2019-05-15 at 19 52 24" src="https://user-images.githubusercontent.com/45913977/57797566-04e6d500-774b-11e9-92aa-46e75272ddda.png">

If you're trying to like a Plume post from Mastodon, type/copy its entire URL into the Mastodon search bar and you should be presented with this:

<img width="1280" alt="Screenshot 2019-05-15 at 19 43 37" src="https://user-images.githubusercontent.com/45913977/57797350-87bb6000-774a-11e9-89e8-0476eb19767f.png">

Clicking 'like' on that post from your Mastodon instance, should result in a like being added on the Plume instance as well:

<img width="1280" alt="Screenshot 2019-05-15 at 19 51 19" src="https://user-images.githubusercontent.com/45913977/57797487-d537cd00-774a-11e9-86f3-ff6945306132.png">

Thus a more detailed description of your steps may be helpful in diagnosing the issue you face.

**Comment by NilsRenaud** [comment]
OK, I tried with your test post on https://fediverse.blog/~/Test/test-article and it works **but it still doesn't work with the article I intended to like** ! 
So I'm suspecting something linked with internationalization since the only difference is that there is special characters in the title (and so in the URL) of this article.

Theses are the exact steps I followed to reproduce the bug : 
- Open firefox 66.0.5 (64 bit) for Ubuntu in private mode
- go to https://fediverse.blog/~/PHBlog/la-quatrième-gratuite
- click on the :heart: button
- I'm directed to https://fediverse.blog/~/PHBlog/la-quatri%C3%A8me-gratuite/remote_interact 
- I fill up the "I'm from another instance" form (which is half English, half French by the way) with my g3l.org account (NilsRenaud@m.g3l.org
![Sélection_001](https://user-images.githubusercontent.com/3686397/57813682-3f636880-7771-11e9-9efb-661842c270bb.png)
- I click on "continue to your instance" button 
- And I end up on this page saying (in french) that something is broken because of an internal error : 
![Sélection_002](https://user-images.githubusercontent.com/3686397/57813727-733e8e00-7771-11e9-8eb8-facbed219602.png)

**Comment by marek-lach** [comment]
Hm... yes that seems to be happening.

**Comment by marek-lach** [comment]
@NilsRenaud @BaptisteGelez Perhaps this is an issue with the URI type in the Rocket Framework itself at the moment?

See: https://github.com/SergioBenitez/Rocket/issues/998

Perhaps this PR will resolve it soon: https://github.com/SergioBenitez/Rocket/pull/941?

**Comment by BaptisteGelez** [comment]
Yes, maybe. I guess we will see once the PR you linked gets merged.
